src/clj/fluree/db/sid.cljs

```diff
@@ -0,0 +1,10 @@
+(ns fluree.db.sid)
+
+(defrecord SID [namespace-code name]
+  IComparable
+  (-compare [_ x]
+    (assert (instance? SID x) "Can't compare an SID to another type")
```

Comment:
I've been wary of having `assert`s at runtime since an `AssertionError` won't get caught by a `(catch Exception ...)` form, so if we end up in this scenario we'll be unwinding the whole stack.

Reply:
This is a good point